Add UI test documentation (#10225)
wordpress-mobile/WordPress-Android: WordPress/src/androidTest/java/org/wordpress/android/README.md

## UI tests Overview

WordPress for Android has UI acceptance tests for critical user flows through the app, such as login, 
signup, and publishing. The tests use mocked network requests with [WireMock](http://wiremock.org/), 
defined in [WordPressMocks](https://github.com/wordpress-mobile/WordPressMocks).

## Running the tests 

**Note that due to the mock server setup, tests cannot be run on physical devices right now.**

1. Follow the [build instructions](https://github.com/wordpress-mobile/WordPress-Android#build-instructions)
 (steps 1-7) to clone, build and run the project in Android Studio.

> 2. Run the tests in `Tests in org.wordpress.android.e2e`

Reviewer: The phrase "tests in" is repeated here. Do you mean:

```suggestion
2. Run the tests in `org.wordpress.android.e2e`
```

> There's no additional steps needed to make the tests run against the mock server -- that's configured set up by default

Reviewer: Should be "configured" or "set up":

```suggestion
There's no additional steps needed to make the tests run against the mock server -- that's configured by default.
```

## Adding a new test?

Great! When adding a new UI test, consider:

* Whether you need to test a user flow (to accomplish a task or goal) or a specific feature (e.g. boundary testing).

> * What screens/pages are being tested(defined in `pages/`).

Reviewer: Nitpick: missing space

```suggestion
* What screens/pages are being tested (defined in `pages/`).
```

> * What user flows are being used(defined in `flows/`).

Reviewer: Nitpick: missing space

```suggestion
* What user flows are being used (defined in `flows/`).
```

* Any specific UI components being interacted with?(defined in `components/`).

> * What network requests are made during the test (defined in the `WordPressMocks` repo).

Reviewer: Since the mocks are included in `libs/mocks/WordPressMocks/` what do you think of referring to that here instead of the `WordPressMocks` repo?

Author: 🤔yeah that'd make more sense to link to it here as well. Something like this? 
```suggestion
* What network requests are made during the test (defined in `libs/mocks/WordPressMocks`)).
```

It's preferred to focus UI tests on entire user flows, and group tests with related flows or goals in the same test file.

> When you add a new test, you may need to add new screens, methods, and flows, when writing these I encourage to check 

Reviewer: Small grammatical fixes:

```suggestion
When you add a new test, you may need to add new screens, methods, and flows. When writing these I encourage you to check 
```

> `support/WPSupportUtils.java` for additional code to reuse to automating the task. Our tests are also used to generate screenshots

Reviewer: Small grammatical fix:

```suggestion
`support/WPSupportUtils.java` for additional code to reuse to automate the task. Our tests are also used to generate screenshots
```

and a lot of useful helper functions are made available through that auxillary file. Wherever possible, avoid the use

>  of a string to select a UI element on the screen, the use of other identifiers such as id or content description is preferable 

Reviewer: Small grammatical fix:

```suggestion
 of a string to select a UI element on the screen; the use of other identifiers such as id or content description is preferable 
```
